files/common/usr/bin/unpack-image

```diff
@@ -113,43 +113,13 @@ fi
 sha256sum -c SHA256SUMS >/dev/null ||
 	die "image is corrupt; checksums don't match"
 
-tar -xzf payload.tar.gz || die "failed to extract payload.tar.gz"
-rm payload.tar.gz || die "failed to remove payload.tar.gz"
+report_progress_inc 50 "Handoff unpack to prepare script."
 
-#
-# We need to be careful when sourcing this file, since it can conflict
-# with (and clobber) functions and/or variables previously defined.
-#
-# shellcheck disable=SC1091
-. version.info || die "sourcing version.info file failed"
-
-[[ -n "$VERSION" ]] || die "VERSION variable is empty"
-[[ -n "$MINIMUM_VERSION" ]] || die "MINIMUM_VERSION variable is empty"
-[[ -n "$MINIMUM_REBOOT_OPTIONAL_VERSION" ]] ||
-	die "MINIMUM_REBOOT_OPTIONAL_VERSION variable is empty"
-
-if $opt_x; then
-	sed -i \
-		"s/^\(MINIMUM_REBOOT_OPTIONAL_VERSION\)=.*$/\1=$VERSION/" \
-		version.info ||
-		die "'sed -i ... version.info' failed"
-
-	# shellcheck disable=SC1091
-	. version.info || die "sourcing version.info file failed"
-fi
+./prepare ${opt_f:+"-f"} ${opt_s:+"-s"} ${opt_x:+"-x"} "$UNPACK_DIR" ||
```

Comment:
How do we want to handle "cleanup" when `prepare` fails?

Do we want `unpack-image` to be responsible for removing `$UNPACK_DIR` when `prepare` fails? Or do we want to `prepare` to be responsible for removing `$UNPACK_DIR` when it fails? Or do we want both to attempt the removal (presumably, one of the removals will fail and/or be a no-op, due to racing with the other)?

Right now, I think both scripts will attempt the removal, but I want to make sure this is by design, rather than by accident.

Reply:
This is by design. No harm in checking if this temp directory still exists after the prepare script returns and clean up in any case that the directory is left behind.